src/lib/RepositoryForms/View/ViewParametersListener.php

```diff
@@ -39,27 +50,53 @@ public static function getSubscribedEvents()
     }
 
     /**
-     * @param PreContentViewEvent $event
+     * @param \eZ\Publish\Core\MVC\Symfony\Event\PreContentViewEvent $event
      */
     public function setViewTemplateParameters(PreContentViewEvent $event)
     {
         $contentView = $event->getContentView();
 
-        if (!$contentView instanceof ContentEditView) {
-            return;
+        if ($contentView instanceof ContentEditView) {
+            /** @var Content $content */
+            $content = $contentView->getParameter('content');
+            $versionInfo = $content->versionInfo;
+            $mainLocationId = $versionInfo->contentInfo->mainLocationId;
+
+            /** Not part of tree yet */
+            if (empty($mainLocationId)) {
+                $parentLocations = $this->locationService->loadParentLocationsForDraftContent($content->versionInfo);
+
+                $contentView->addParameters([
+                    'parentLocations' => $parentLocations,
+                ]);
+            }
+
+            $this->processCreator($versionInfo, $contentView);
         }
 
-        /** @var Content $content */
-        $content = $contentView->getParameter('content');
-        $mainLocationId = $content->versionInfo->contentInfo->mainLocationId;
+        if ($contentView instanceof UserUpdateView) {
```

Comment:
Please create separate methods for User and Content and attach both to the same event name. This makes methods less bloated.